src/dotty/runtime/Arrays.scala

```diff
@@ -22,35 +24,8 @@ object Arrays {
     arr
   }
 
-  /** Create an array of type T. T must be of form Array[E], with
-   *  E being a reference type.
+  /** Create an array of a reference type T.
    */
-  def newRefArray[T](length: Int): T = ???
-
-  /** Create a Byte[] array */
-  def newByteArray(length: Int): Array[Byte] = ???
-
-  /** Create a Short[] array */
-  def newShortArray(length: Int): Array[Short] = ???
-
-  /** Create a Char[] array */
-  def newCharArray(length: Int): Array[Char] = ???
-
-  /** Create an Int[] array */
-  def newIntArray(length: Int): Array[Int] = ???
-
-  /** Create a Long[] array */
-  def newLongArray(length: Int): Array[Long] = ???
-
-  /** Create a Float[] array */
-  def newFloatArray(length: Int): Array[Float] = ???
-
-  /** Create a Double[] array */
-  def newDoubleArray(length: Int): Array[Double] = ???
-
-  /** Create a Boolean[] array */
-  def newBooleanArray(length: Int): Array[Boolean] = ???
-
-  /** Create a scala.runtime.BoxedUnit[] array */
-  def newUnitArray(length: Int): Array[Unit] = ???
+  def newArray[Arr](componentType: Class[_], returnType: Class[Arr], dimensions: Array[Int]): Arr =
```

Comment:
That's the intention!
If `dimension.size ==1` backend will create a normal array
If `dimension.size > 1` backend will issue `multianewarray` instruction.

See example here: https://github.com/dotty-staging/dotty/blob/remove-newarray-magic/tests/run/MultiArr.scala